File: app/src/main/java/org/cuieney/videolife/presenter/contract/EssayHomeDetailContract.java
package org.cuieney.videolife.presenter.contract;

import org.cuieney.videolife.common.base.BasePresenter;
import org.cuieney.videolife.common.base.BaseView;
import org.cuieney.videolife.entity.EssayDetailBean;

import java.util.List;

/**
 * Created by cuieney on 2017/5/17.
 */

public interface EssayHomeDetailContract {
    interface View extends BaseView {
        void showContent(EssayDetailBean dataBeanList);
        void error(Throwable throwable);
    }

    interface Presenter extends BasePresenter<EssayHomeDetailContract.View> {
        void getEssayDetailData(int id,int sourceId);

    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/contract/EssayHomeDetailContract.kt
package org.cuieney.videolife.kotlin.presenter.contract

import org.cuieney.videolife.entity.EssayDetailBean
import org.cuieney.videolife.entity.YiRenBean.DataBean
import org.cuieney.videolife.kotlin.base.BasePresenter
import org.cuieney.videolife.kotlin.base.BaseView

/**
 * Created by cuieney on 2017/5/23.
 */

interface EssayHomeDetailContract{
    interface View : BaseView {
        fun showContent(essayDetail: EssayDetailBean)
        fun error(throwable: Throwable)
    }

    interface Presenter : BasePresenter {
         fun getEssayDetailData(id: Int, sourceId: Int,view:View)
    }
}
